src/components/select-field/Overlay.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import { getDimensions, getViewportIntersections } from '../../utils/dom';
+import { OVERLAY_WRAPPER_CLASS, VIEWPORT_BORDERS } from '../../constants';
+
+type Props = {
+    children: React.Node,
+    className?: string,
+    isOpen?: boolean,
+};
+
+type Dimensions = {
+    height?: number,
+    width?: number,
+};
+
+type Translations = {
+    x?: number,
+    y?: number,
+};
+
+function getDimensionsWhileHidden(element: HTMLElement): Dimensions {
+    if (!element) {
+        return {};
+    }
+
+    element.classList.add('offscreen');
+    const dimensions = getDimensions(element);
+    element.classList.remove('offscreen');
```

Comment:
What is the purpose of adding and removing class `offscreen` on either side of getting element's dimensions?

Reply:
I want to get the dimensions of the element while not visible, so adding the `offscreen` class allows me to do so. But I think I am changing up the CSS so that I can avoid this